### ISSUE TITLE
поле Quantity в проводках по складским док-там

### ISSUE THREAD

**Issue by alexandra-gsoftware** [GoldenSoftwareLtd/gedemin#3775]
[L.QUANTITY] конвертируется в gdcDocument.FieldByName("QUANTITY").Value. Но разве Количество не может быть пустым?

**Comment by alexandra-gsoftware** [comment]
да и не только оно... from и to признаки типа dcurrency тоже.

**Comment by NikolayUkleyko** [comment]
> _”Но разве Количество не может быть пустым?”_

  Может быть пустым.

> _”from и to признаки типа dcurrency тоже.”_

  @alexandra-gsoftware , пожалуйста, уточни.

**Comment by gsbelarus** [comment]
Новейший экзешник взять (только не один файл, а весь архив). И опять придется переоткрыть/пересохранить все ТХО/ТП. (Пусть Карпуку икается за то, что он сделал это окно не в MVC).

**Comment by alexandra-gsoftware** [comment]
Николай, я имела в виду, что эти поля могут быть пустыми, но в проводках обрабатываются так, будто бы не могут

**Comment by NikolayUkleyko** [comment]
@gsbelarus что значит

> _”только не один файл, а весь архив ?”_

  Были изменения в дополнительных файлах для gedemin.exe ? Если, да, то, какие ?

**Comment by alexandra-gsoftware** [comment]
все тп или только по позициях складских док-тов?

**Comment by gsbelarus** [comment]
нет. то что я менял не помогло. @MIchaelShoihet  нужен пример и нужно искать где это формируется.

**Comment by alexandra-gsoftware** [comment]
если поле участвует в сравнении или расчетах или присвоении переменной, то оно должно быть AsCurrency. Value - только когда присваиваем аналитику типа DCURRENCY (если такие вдруг есть, потому что на эталоне нет), или когда оно (поле) проверяется ф-ей IsNull()

**Comment by alexandra-gsoftware** [comment]
@ilonaand , а ты что скажешь?

**Comment by alexandra-gsoftware** [comment]
Илона согласна:)

**Comment by alexandra-gsoftware** [comment]
вот кстати при перекомпиляции от 01.03.2016 строка выглядит так:
      gdcEntry.FieldByName("creditncu").AsCurrency = gdcDocument.FieldByName("OUTQUANTITY").AsCurrency \* gdcDocument.FieldByName("FROM_USR$INV_COSTACCNCU").AsCurrency

(это очень даже правильно)

А сегодня - вот так:
      gdcEntry.FieldByName("creditncu").AsCurrency = gdcDocument.FieldByName("OUTQUANTITY").Value \* gdcDocument.FieldByName("FROM_USR$INV_COSTACCNCU").Value

**Comment by NikolayUkleyko** [comment]
@alexandra-gsoftware вопрос, если такие изменения (Value > AsCurrency), то как оно вообще работало корректно до сих пор ?

**Comment by alexandra-gsoftware** [comment]
оно не так давно поломалось. еще 1го марта были карренси.

**Comment by alexandra-gsoftware** [comment]
нужно перекомпилировать проводки? Или все вернулось на версию 01.03.2016?

**Comment by gsbelarus** [comment]
желательно перекомпилировать

**Comment by alexandra-gsoftware** [comment]
с типом currency сейчас все хорошо, но появилось зато вот такое:

If gdcDocumentHeader.FieldByName("USR$CHECKSALE").Value = 1 Then

а раньше оно выглядело как

If gdcDocumentHeader.FieldByName("USR$CHECKSALE").AsInteger = 1 Then

Старый вариант лучше.

**Comment by alexandra-gsoftware** [comment]
Встречается в следующих местах:
AutoEntryScript147080154_23783130_line (Торговля - метаданные)
AutoEntryScript147235098_597341771_line (Учет в автопарке - Метаданные)
Sub_1_147037889_81366384 (Проводки материального склада)
Sub_1_147092182_106202483 (Торговля - метаданные)
Sub_2_147080295_23783130 ( Торговля - метаданные)
TrEntryScript147237413_597341771 (Учет в автопарке - ТХО)
TrEntryScript147272315_820036456 (Учет в автопарке - ТХО)
AutoEntryScript147021480_157887963_line ( Рублевые банковские документы 10)
AutoEntryScript147044743_81366384_line (Мат_склад - метаданные)
AutoEntryScript147044747_81366384_line (Мат_склад - метаданные)
AutoEntryScript147080082_23783130_line ( Торговля - метаданные)
AutoEntryScript147308500_827477687_line (Настройка банковской выписки и картотеки)
Sub_1_147079921_23783130 ( Торговля - метаданные)
Sub_1_147092179_106202483 (Торговля - метаданные)
Sub_1_147092183_106202483 (Торговля - метаданные)
TrEntryScript147254011_1846313658 (Акт выполненных работ)
TrEntryScript148966782_57856281( Акт выполненных работ)
TrEntryScript330267559_1463738259 ( Акт выполненных работ)

**Comment by NikolayUkleyko** [comment]
Коллеги, если, на ,ваш, взгляд данная тема уже неактуальна, пожалуйста, закройте ее.
@alexandra-gsoftware , пожалуйста, изложи свою точку зрения.

**Comment by NikolayUkleyko** [comment]
Проверено.
Работает корректно.
